src/HttpCache/SymfonyPurger.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\HttpCache;
+
+use Toflar\Psr6HttpCacheStore\Psr6StoreInterface;
```

Comment:
Wouldn't it be possible to move this new implementation directly in Symfony?

Reply:
No, I tried and it was denied. See https://github.com/symfony/symfony/pull/20061 for the whole discussion.

Reply:
Ok... I'm not confident with adding (soft) dependencies to non-popular yet libraries (like `Psr6HttpCacheStore`). We already had problems with bridges to 3rd party libraries not very well maintained (FOS things...).

My proposal is:
* Tweak the config in this PR to allow to easily register external Purgers from the config (using a service name for instance).
* Move the `SymfonyPurger` class and the trait in an external bundle (or directly in `Psr6HttpCacheStore`) for now

WDYT?

Reply:
Yeah I understand. In that case, it's my library and it consists of one class so I can basically guarantee it's maintained because I can't think of a lot of stuff that would need to be maintained there 😄 But anyway, I just feel it can never get popular if we don't give it the chance. I mean, I'm totally okay with putting the APIP integration into it's own bundle if you don't want it in core but then it will be just another one floating around in my personal namespace instead of APIP so chances are, people won't even notice there's such a thing.

Reply:
It's mostly a docs issue to me. I'm totally ok to add a new doc entry explaining how to install your lib (that will be required to install manually anyway) to get cache invalidation in pure PHP.

Reply:
:+1: 

@Toflar just add the SymfonyPurger and the StoreAwareKernelInterface to your repository! 
Patch api platform to allow the use of more then 1 purger in this pull request. 

Then we just add a doc entry! I'd be the first to enjoy the pure PHP cache I think it's a great idea!

Reply:
The Kernel should be in whatever namespace.

If the purger is implementing our PurgerInterface yeah it can be in a Bridge/ApiPlatform thingy

Reply:
Okay, no problem. There's one thing that needs to be clarified first though: the encoding of the tags. I see two options here:

1) I add a new interface every purger can implement that allows to ask the purger to encode the tags whereas fallback would be the current, comma-separated list. Pros: Very flexible, Cons: more complicated.

2) We use `md5` or `sha1` by default and add a `Cache-Tags-Debug` header which shows  the raw tags if `kernel.debug` is enabled. I cannot think of any cache implementation that cannot handle hashes as keys. Pros: Simple, Cons: Less flexible.

WDYT?

Reply:
I'd go for option 1

Reply:
Okay, would you still add the additional debug header? I think it might be very useful and it doesn't harm because it's hidden in production anyway.

Reply:
Sure I think it's a good idea as well!